inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/ActiveLearningServiceImpl.java

```diff
@@ -102,4 +114,339 @@ public ActiveLearningServiceImpl(DocumentService aDocumentService,
         return model.getFlattenedPredictions(aState.getDocument().getName(), aSelectedLayer,
             windowBegin, windowEnd, aJcas, true);
     }
+
+    @Override
+    public void putSessionActive(User aUser, Project aProject, boolean aSesscionActive)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        synchronized (state) {
+            state.setSessionActive(aProject, aSesscionActive);
+        }
+    }
+
+    @Override
+    public boolean getSessionActive(User aUser, Project aProject)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        boolean sessionActive;
+        synchronized (state) {
+            sessionActive = state.isSessionActive(aProject);
+        }
+        return sessionActive;
+    }
+
+    @Override
+    public void putHasUnseenRecommendation(User aUser, Project aProject,
+        boolean aHasUnseenRecommendation)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        synchronized (state) {
+            state.setHasUnseenRecommendation(aProject, aHasUnseenRecommendation);
+        }
+    }
+
+    @Override
+    public boolean getHasUnseenRecommendation(User aUser, Project aProject)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        boolean hasUnseenRecommendation;
```

Comment:
I think there is no need for the variable, just `return` from inside the synchronized block (applies to other methods as well).